static/app/views/starfish/views/spans/spanSummaryPanel/useSpanTransactionMetrics.tsx

```diff
@@ -0,0 +1,51 @@
+import keyBy from 'lodash/keyBy';
+
+import {useQuery} from 'sentry/utils/queryClient';
+import {HOST} from 'sentry/views/starfish/utils/constants';
+import type {Span} from 'sentry/views/starfish/views/spans/spanSummaryPanel/types';
+
+const INTERVAL = 12;
+
+type Metric = {
+  p50: number;
+  spm: number;
+};
+
+export const useSpanTransactionMetrics = (
+  span?: Span,
+  transactions?: string[],
+  referrer = 'span-transaction-metrics'
+) => {
+  const query =
+    span && transactions && transactions.length > 0 ? getQuery(span, transactions) : '';
+
+  const {isLoading, error, data} = useQuery<Metric[]>({
+    queryKey: [
+      'span-transactions-metrics',
+      span?.group_id,
+      transactions?.join(',') || '',
+    ],
+    queryFn: () =>
+      fetch(`${HOST}/?query=${query}&referrer=${referrer}`).then(res => res.json()),
+    retry: false,
+    initialData: [],
+    enabled: Boolean(query),
+  });
+
+  const parsedData = keyBy(data, 'transaction');
+
+  return {isLoading, error, data: parsedData};
+};
```

Comment:
Probs doesn't matter long term as its experimental. 
But what if you did this, as it would retain all the other properties of useQuery such is `isRefetching`
```suggestion
  const result = useQuery<Metric[]>({
    queryKey: [
      'span-transactions-metrics',
      span?.group_id,
      transactions?.join(',') || '',
    ],
    queryFn: () =>
      fetch(`${HOST}/?query=${query}&referrer=${referrer}`).then(res => res.json()),
    retry: false,
    initialData: [],
    enabled: Boolean(query),
  });

  const parsedData = keyBy(result.data, 'transaction');

  return {...result, data: parsedData} as DefinedUseQueryResult<Record<string, {p50: number; spm: number}>>;
};
```

Reply:
Oh that's a good point! I'll make that change in an upcoming PR, thanks!

Reply:
np! fyi i'm not sure if I typed it properly in my suggestion, so make sure to double check that's all good.